CLI/src/CLI.cs

```diff
@@ -272,7 +272,8 @@ private void AddProgramsToRootTable(SymbolTable rootTable, IEnumerable<Program>
                     var previousPrograms = rootTable.GetPrograms();
                     foreach (var previousProgram in previousPrograms)
                     {
-                        previousProgram.SymbolTable.GetTableFromScope(SymbolTable.Scope.Namespace).AddProgram(program);
+                        var namespaceSymbolTable = previousProgram.SymbolTable.GetTableFromScope(SymbolTable.Scope.Namespace);
```

Comment:
This certainly fixes the bug but could you investigate why it is needed to add programs into namespace SymbolTable and why sometimes it creates duplicates ?

Reply:
- It is needed to add programs main and stacked (nested are inside symbolTable of program) into Namespace-SymbolTable, for each program, even from different file, so parser can understand call of something inside another program.
- Just after CUP analyse, for each main and stacked program, ProgramClassbuilder creates a program with a TableOfNamespaces associated.
TableOfNamespaces (SymbolTable with scope Namespace) contains all the programs already analyzed, at this point. Thus, the SymbolTable of scope Namespace of a stacked program contains a more or less complete list of programs.
- This is why sometimes it creates duplicates in case of stacked program.

modification AddProgramsToRootTable and SymbolTable in order managing programs with same name
https://github.com/TypeCobolTeam/TypeCobol/pull/1650/commits/aee6bdd991f556ad40e12637d1f12234efcbefe6

Reply:
Althouhg I don't get why we get duplicates only for stacked, I think we can improve this part but it requires a bit of refactoring of the `CLI` class. Instead of having multiple instances of parsers and multiple Namespace SymbolTable, we could use the same SymbolTable all the way. We'll discuss about that.

Reply:
Stacked programs are already added to namespace SymbolTable by ProgramClassBuilder, so I think that's why we get duplicate.

To fix this bug, we must first clarify if a stacked program can be referenced from another source file.
From a TypeCobol perpsective:
- We don't need stacked program because we already have procedure.
- Our generated code doesn't works for procedure inside stacked program

As mentionned by Lanoydo, we still need to handle the case of 2 programs with the same name.

Reply:
Code has been modify by no longer filling the symbol table with programs stacked from another source file.
https://github.com/TypeCobolTeam/TypeCobol/pull/1650/commits/7562492ef5d810d8b308bdc276630234305d83b8

Reply:
Modify test  to verify handling the case of 2 programs with the same name
https://github.com/TypeCobolTeam/TypeCobol/pull/1650/commits/8757ce7ea03a9b960648331204bc55bc744e224e

Reply:
After discussion with @sarterbe, stacked cannot be call by others programs in our environment.
So there is no need to support public procedure in stacked programs.

Reply:
done.
https://github.com/TypeCobolTeam/TypeCobol/pull/1650/commits/3680098497555087b762ae698ab7cf256eb2db6a